### PR TITLE
Implement graphql types and endpoints

### PR DESCRIPTION
**Pull request by delkopiso**
- Use a separate unauthenticated schema for authentication
- Add request specs to cover GraphQL queries and mutations

### CODE REVIEW

File: app/controllers/api/auth_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module Api
+  class AuthController < ApplicationController
+    def execute
+      result = Graph::AuthSchema.execute(
+        params[:query],
+        context: { current_user: current_user }
```
Comment (farisj) [inline app/controllers/api/auth_controller.rb]: gonna leave this controller to @Ewarren7 to review

File: app/models/user.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ class User < ActiveRecord::Base
   validates :email, presence: true
   validates :avatar_url, presence: true
   validates :github_id, presence: true
-  # validates :api_token, presence: true
+  validates :api_token, presence: true
```
Comment (farisj) [inline app/models/user.rb]: I'm wondering what will happen with this validation. What's the status on this field now?

Also, Is there a Lion side PR to this work that's using these endpoints?

**Comment by farisj** [comment]
Aside from question about the validation, LGTM

File: app/models/user.rb (hunk 2 above)
Comment (delkopiso) [inline app/models/user.rb]: The validation should be active in production. Being commented out was an oversight.

Comment (delkopiso) [inline app/models/user.rb]: The lion PR is https://github.com/alphasights/lion/pull/33

**Pull request merged**